src/core/electrostatics_magnetostatics/dipole_inline.hpp

```diff
@@ -8,33 +8,36 @@
 
 namespace Dipole {
 // forces_inline
-inline void calc_pair_force(Particle *p1, Particle *p2,
-                            Utils::Vector3d const &d, double dist, double dist2,
-                            Utils::Vector3d &force) {
+inline std::tuple<Utils::Vector3d, Utils::Vector3d, Utils::Vector3d>
+calc_pair_force(Particle const *const p1, Particle const *const p2,
+                Utils::Vector3d const &d, double dist, double dist2) {
+  Utils::Vector3d force{}, torque1{}, torque2{};
   switch (dipole.method) {
 #ifdef DP3M
   case DIPOLAR_MDLC_P3M:
     // fall trough
   case DIPOLAR_P3M: {
+    double eng;
+    std::tie(eng, force, torque1, torque2) =
```

Comment:
if there was not this energy part in the returned tuple we could just do ```return dp3m_add_pair_force(p1, p2, d, dist2, dist)```

Reply:
that's inelegant for sure, but we need the energy for NPT on the next line, and reducing the use of output parameters is our goal. I don't think there is a performance impact (https://godbolt.org/z/drOVeJ).